packages/errors/src/context.ts

```diff
@@ -38,95 +95,166 @@ export type DefaultUnspecifiedErrorContextToUndefined<T> = {
-    };
-    [SOLANA_ERROR__CODECS_CANNOT_DECODE_EMPTY_BYTE_ARRAY]: {
-        codecDescription: string;
-    };
-    [SOLANA_ERROR__CODECS_FIXED_SIZE_ENCODER_DECODER_SIZE_MISMATCH]: {
-        decoderFixedSize: number;
-        encoderFixedSize: number;
-    };
-    [SOLANA_ERROR__CODECS_VARIABLE_SIZE_ENCODER_DECODER_MAX_SIZE_MISMATCH]: {
-        decoderMaxSize: number | undefined;
-        encoderMaxSize: number | undefined;
-    };
-    [SOLANA_ERROR__CODECS_WRONG_NUMBER_OF_BYTES]: {
-        bytesLength: number;
-        codecDescription: string;
-        expected: number;
-    };
-    [SOLANA_ERROR__EXPECTED_DECODED_ACCOUNT]: {
-        address: string;
-    };
-    [SOLANA_ERROR__FAILED_TO_DECODE_ACCOUNT]: {
-        address: string;
-    };
-    [SOLANA_ERROR__INCORRECT_BASE58_ADDRESS_BYTE_LENGTH]: {
-        actualLength: number;
-    };
-    [SOLANA_ERROR__INCORRECT_BASE58_ADDRESS_LENGTH]: {
-        actualLength: number;
-    };
-    [SOLANA_ERROR__INVALID_KEYPAIR_BYTES]: {
-        byteLength: number;
-    };
-    [SOLANA_ERROR__MAX_NUMBER_OF_PDA_SEEDS_EXCEEDED]: {
-        actual: number;
-        maxSeeds: number;
-    };
-    [SOLANA_ERROR__MAX_PDA_SEED_LENGTH_EXCEEDED]: {
-        actual: number;
-        index: number;
-        maxSeedLength: number;
-    };
-    [SOLANA_ERROR__MULTIPLE_ACCOUNTS_NOT_FOUND]: {
-        addresses: string[];
-    };
-    [SOLANA_ERROR__NONCE_ACCOUNT_NOT_FOUND]: {
-        nonceAccountAddress: string;
-    };
-    [SOLANA_ERROR__NONCE_INVALID]: {
-        actualNonceValue: string;
-        expectedNonceValue: string;
-    };
-    [SOLANA_ERROR__NOT_A_BASE58_ENCODED_ADDRESS]: {
-        putativeAddress: string;
-    };
-    [SOLANA_ERROR__NOT_ALL_ACCOUNTS_DECODED]: {
-        addresses: string[];
-    };
-    [SOLANA_ERROR__PROGRAM_DERIVED_ADDRESS_BUMP_SEED_OUT_OF_RANGE]: {
-        bump: number;
-    };
-    [SOLANA_ERROR__RPC_INTEGER_OVERFLOW]: {
-        argumentLabel: string;
-        keyPath: readonly (string | number | symbol)[];
-        methodName: string;
-        optionalPathLabel: string;
-        path?: string;
-        value: bigint;
-    };
-    [SOLANA_ERROR__TRANSACTION_MISSING_SIGNATURES]: {
-        addresses: string[];
-    };
-    [SOLANA_ERROR__TRANSACTION_ERROR_UNKNOWN]: {
-        errorName: string;
-        transactionErrorContext?: unknown;
-    };
-    [SOLANA_ERROR__TRANSACTION_ERROR_DUPLICATE_INSTRUCTION]: {
-        index: number;
-    };
-    [SOLANA_ERROR__TRANSACTION_ERROR_INSUFFICIENT_FUNDS_FOR_RENT]: {
-        accountIndex: number;
-    };
-    [SOLANA_ERROR__TRANSACTION_ERROR_PROGRAM_EXECUTION_TEMPORARILY_RESTRICTED]: {
-        accountIndex: number;
-    };
-}>;
+export type SolanaErrorContext = DefaultUnspecifiedErrorContextToUndefined<
+    BasicInstructionErrorContext<
+        | typeof SOLANA_ERROR__INSTRUCTION_ERROR_UNKNOWN
+        | typeof SOLANA_ERROR__INSTRUCTION_ERROR_GENERIC_ERROR
+        | typeof SOLANA_ERROR__INSTRUCTION_ERROR_INVALID_ARGUMENT
```

Comment:
Why is this needed now?

Reply:
I wanted a utility that unconditionally added `index` to _all_ the instruction error contexts.